### PR TITLE
Onnx support: add pos parameter to softmax

### PR DESCRIPTION
**Pull request by lingxiao1989**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this patch)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If it is possible, please attach a screenshot; otherwise, remove this)

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala [hunk 1]
```diff
@@ -31,61 +31,74 @@ import scala.reflect.ClassTag
...
+  }
+
   override def updateOutput(input: Tensor[T]): Tensor[T] = {
     require(1 <= input.nDimension() && input.nDimension() <= 4,
       "1D, 2D, 3D or 4D tensor expected" +
         s"input dimension ${input.nDimension()}")
-    val (nFrame, stride) = if (input.nDimension() == 1) {
+    pos = getPositiveDimension(input)
+
+    val (nFrame, stride) = if (input.nDimension() - pos == 0) {
```
[the remark this answers is not in the document]
Comment (LeicongLi) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala]: And add comments.
It seems we are trying to calculate the stride size, but its not straightforward.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala [hunk 2]
```diff
@@ -31,61 +31,74 @@ import scala.reflect.ClassTag
...
       results = new Array[Future[Unit]](nFrame * stride)
     }
     output.resizeAs(input)
-    SoftMax.updateOutput[T](input, output, results)
+    SoftMax.updateOutput[T](input, output, results, pos)
     output
   }
-
   override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
+    pos = getPositiveDimension(input)
```
Comment (LeicongLi) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala]: pos should be computed only once, not every time we do updates.

**Comment by LeicongLi** [comment]
Please check related python api, sync the change if necessary.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (lingxiao1989) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala]: ok

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala (hunk 2 above)
Comment (lingxiao1989) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SoftMax.scala]: ok, I am not sure whether the updateGradInput will always be called after updateOutput, so I added getPositiveDimension in both of them.

**Comment by lingxiao1989** [comment]
Jenkins:
http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/962/

**Pull request merged**